superset/views/core.py

```diff
@@ -256,6 +256,13 @@ def pre_add(self, db):
     def pre_update(self, db):
         self.pre_add(db)
 
+    def pre_delete(self, obj):
+        if obj.tables:
+            raise SupersetException(Markup(
+                'Cannot delete a database that has tables attached. '
+                "Here's the list of associated tables: " +
+                ''.join(['{}'.format(o) for o in obj.tables])))
```

Comment:
You want to join them with a comma here, no? Also, you don't need format nor the list comprehension:

```python
', '.join(str(o) for o in obj.tables)))
```

Reply:
Ah, true... I keep forgetting about Python 2.

And yeah, removing the brackets create a generator which is immediately consumed by the `join` method.